python/cudf/cudf/core/dataframe.py

```diff
@@ -2602,6 +2602,71 @@ def insert(self, loc, name, value, nan_as_null=None):
 
         self._data.insert(name, value, loc=loc)
 
+    def diff(self, periods=1, axis=0):
```

Comment:
pandas verifies that `period` is an integer. We may want to perform that check? https://github.com/pandas-dev/pandas/blob/42a26ac539ac9a194f6e7da0897ffc938b7d7e76/pandas/core/frame.py#L8521-L8528